.cci.jenkinsfile

```diff
@@ -3,20 +3,56 @@
 cosaPod(buildroot: true) {
     checkout scm
 
-    // we don't need the qemu image to test coreos-installer; just the OSTree
-    fcosBuild(make: true, skipKola: true, extraArgs: 'ostree')
-
-    stage("Build metal+live") {
-        shwrap("cd /srv/fcos && cosa buildextend-metal")
-        shwrap("cd /srv/fcos && cosa buildextend-metal4k")
-        // Compress once
-        shwrap("cd /srv/fcos && cosa buildextend-live")
-        shwrap("cd /srv/fcos && cosa compress --artifact=metal --artifact=metal4k")
-    }
-    stage("Test ISO") {
-        // No need to run the iso-live-login scenario (in theory, and also right
-        // now it's buggy)
-        shwrap("cd /srv/fcos && kola testiso -S --scenarios pxe-install,iso-install,iso-offline-install")
-        shwrap("cd /srv/fcos && kola testiso -S --scenarios iso-install --qemu-native-4k")
+    // so cargo doesn't use /.cargo, which it can't access
+    withEnv(["CARGO_HOME=${env.WORKSPACE}/.cargo"]) {
+        shwrap("make")
     }
+
+    parallel(
+        lint: stage("Lint") {
+            // Check help text maximum line length
+            shwrap('''
+                checklen() {
+                    local length
+                    length=$(target/debug/coreos-installer $* --help | wc -L)
+                    if [ "${length}" -gt 80 ] ; then
+                        echo "$* --help line length ${length} > 80"
+                        fail=1
+                    fi
+                }
+                checklen
+                checklen install
+                checklen download
+                checklen list-stream
+                checklen iso
+                checklen iso embed
+                checklen iso show
+                checklen iso remove
+                if [ "${fail}" = 1 ]; then
```

Comment:
Hmm, I think we need to use `${fail:-}` or otherwise initialize it to `fail=0` at the start.

Reply:
(Because otherwise I think `set -u` will trip on this in the happy case.)

Reply:
Yup, I'm not used to `set -u`.